strawberryfields/io/xir_io.py

```diff
@@ -13,47 +13,77 @@
 # limitations under the License.
 """
 This module contains functions for loading and saving Strawberry Fields
-:class:`~.Program` objects from/to Blackbird scripts and Strawberry Fields
+:class:`~.Program` objects from/to XIR scripts and Strawberry Fields
 code.
 """
 # pylint: disable=protected-access,too-many-nested-blocks
 from decimal import Decimal
-from typing import Iterable
+from typing import Iterable, List
 
 import numpy as np
 
 import xir
-from xir.program import Declaration
 
-import strawberryfields.program as sfp
-from strawberryfields.tdm.tdmprogram import TDMProgram
 import strawberryfields.parameters as sfpar
+from strawberryfields.program import Program
+from strawberryfields.tdm.tdmprogram import TDMProgram, is_ptype
 from strawberryfields import ops
 
 
+def get_expanded_statements(prog: xir.Program) -> List[xir.Statement]:
+    """Get a list of statements with all gate definitions expanded.
+
+    Args:
+        prog (xir.Program): XIR program with statements and definitions
+
+    Returns:
+        list[xir.Statement]: list of expanded XIR statements
+    """
+    statements = []
+    for op in prog.statements:
+        sub_statements = prog.gates.get(op.name, [])
```

Comment:
No need to default to the empty list here:
```suggestion
        sub_statements = prog.gates.get(op.name)
```

Reply:
Leftover from before I had the `if sub_statements` and iterated directly over `sub_statements`.